### DISCUSSION TITLE
naming dynamically computed f.expression in by

### DISCUSSION THREAD

**Discussion by lmullany** [h2oai/datatable#3219] [Q&A] [answered]
Is it possible to name the dynamically computed column in `by()`?

```python
from datatable import dt, f, by
df = dt.Frame({"group1": ["A", "A", "B", "B", "A"],
               "group2": [1, 0, 1, 1, 1],
               "var1": [343, 345, 567, 345, 212]})
df[:, {"total":f.var1.sum()},by(f.group1=="A")]
```
This returns
```
	C0	total
▪	▪▪▪▪▪▪▪▪
0	0	912
1	1	900
```
i.e. rather than `C0`, can we name it as its being computed?.. Something similar to R, where we can do
```R
df[, .("total" = sum(var1)), by=.(grpA = group1=="A")]
```

**Comment by samukweku** [comment, marked as answer]
There is a [feature request](https://github.com/h2oai/datatable/issues/2504) for this; in one of the discussion threads, @st-pasha suggested the keyword `as`, similar to that in SQL. No implementation yet.

**Reply by samukweku** [reply]
`.alias` handles this now
